openxr/src/passthrough.rs

```diff
@@ -0,0 +1,226 @@
+use crate::{
+    cvt, raw, session, sys, PassthroughFlagsFB, PassthroughLayerPurposeFB, Result, Session,
+    SessionInner,
+};
+use std::ptr;
+use std::sync::Arc;
+use sys::PassthroughLayerFB;
+
+/// A [passthrough feature].
+///
+/// Requires [`XR_FB_passthrough`] to be enabled.
+///
+/// See the [`PassthroughFB struct`].
+///
+/// [`XR_FB_passthrough`]: https://www.khronos.org/registry/OpenXR/specs/1.0/html/xrspec.html#XR_FB_passthrough
+/// [`PassthroughFB struct`]: https://docs.rs/openxr-sys/latest/openxr_sys/struct.PassthroughFB.html
+/// [passthrough feature]: https://developer.oculus.com/documentation/native/android/mobile-passthrough/#create-and-start-a-passthrough-feature
+pub struct Passthrough {
+    pub(crate) session: Arc<session::SessionInner>,
+    handle: sys::PassthroughFB,
+}
+
+impl Passthrough {
+    /// [Create](https://www.khronos.org/registry/OpenXR/specs/1.0/man/html/openxr.html#xrCreatePassthroughFB) a [passthrough feature].
+    ///
+    /// # Arguments
+    ///
+    /// * `flags` - the passthrough feature flags. Unless `PassthroughFlagsFB::IS_RUNNING_AT_CREATION` is enabled, `Self::start` must be called.
+    ///
+    /// # Errors
+    ///
+    /// This function return `ERROR_EXTENSION_NOT_PRESENT` if [`XR_FB_passthrough`] is disabled.
+    ///
+    /// [`XR_FB_passthrough`]: https://www.khronos.org/registry/OpenXR/specs/1.0/html/xrspec.html#XR_FB_passthrough
+    /// [passthrough feature]: https://developer.oculus.com/documentation/native/android/mobile-passthrough/#create-and-start-a-passthrough-feature
+    pub(crate) fn create<G>(session: &Session<G>, flags: PassthroughFlagsFB) -> Result<Self> {
+        let info = sys::PassthroughCreateInfoFB {
+            ty: sys::PassthroughCreateInfoFB::TYPE,
+            next: ptr::null(),
+            flags,
+        };
+        let fp = session
+            .inner
+            .instance
+            .exts()
+            .fb_passthrough
+            .as_ref()
+            .ok_or(sys::Result::ERROR_EXTENSION_NOT_PRESENT)?;
+        let mut handle = sys::PassthroughFB::NULL;
+        unsafe {
+            cvt((fp.create_passthrough)(
+                session.as_raw(),
+                &info,
+                &mut handle,
+            ))?;
+        }
+        Ok(Passthrough {
+            session: session.inner.clone(),
+            handle,
+        })
+    }
+
+    /// [Pause](https://www.khronos.org/registry/OpenXR/specs/1.0/man/html/openxr.html#_xrpassthroughstartfb3) a [passthrough feature].
+    ///
+    /// # Errors
+    ///
+    /// This function return `ERROR_EXTENSION_NOT_PRESENT` if [`XR_FB_passthrough`] is disabled.
+    ///
+    /// [`XR_FB_passthrough`]: https://www.khronos.org/registry/OpenXR/specs/1.0/html/xrspec.html#XR_FB_passthrough
+    /// [passthrough feature]: https://developer.oculus.com/documentation/native/android/mobile-passthrough/#create-and-start-a-passthrough-feature
+    pub(crate) fn pause(&mut self, session: &Arc<SessionInner>) -> Result<()> {
```

Comment:
The `session` argument is redundant, as the appropriate session is already stored inside `self`.

Reply:
Correct, TY!